src/main/groovy/com/wooga/gradle/io/FileUtils.groovy

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2018 Wooga GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.wooga.gradle.io
+
+import org.gradle.api.file.RegularFileProperty
+
+trait FileUtilsImpl {
+
+    /**
+     * If the file does not exist at the given path, it creates it by
+     * ensuring any missing directories along the path are also created.
+     * @param file The file to be ensured
+     */
+    static void ensureFile(File file) {
+        if(!file.exists()) {
+            File parent = file.getParentFile()
+            if (!parent.exists() && !parent.mkdirs()) {
+                throw new IllegalStateException("Couldn't create dir: " + parent)
+            }
+            file.createNewFile()
+        }
+    }
+
+    /**
+     * If the file does not exist at the given path, it creates it by
+     * ensuring any missing directories along the path are also created.
+     * @param file The file to be ensured
+     */
+    static void ensureFile(RegularFileProperty file) {
+        ensureFile(file.get().asFile)
+    }
+}
+
+class FileUtils implements FileUtilsImpl {
+}
+
```

Comment:
This makes no sense for a trait. I know one would have the static method without import. But I would create a simple FileUtils class with static methods. I'm still not sure what happens when the trait gets implemented multiple times.

Reply:
Also check out [commons-math](https://commons.apache.org/proper/commons-io/apidocs/index.html)
It would be overkill here but something to keep in mind before adding more and more methods.

Reply:
I agree with it not being a trait.

Reply:
I believe this was a vestige of way back while developing the atlas-unity test classes that I forgot to remove.